### PR TITLE
feat(helm-manager): allow either tag or version for images in helm charts

### PR DESCRIPTION
**Pull request by karl-johan-grahn**
## Changes

Allow image versions in Helm charts to be specified either by using `version` or `tag`.

## Documentation (please check one with an [x])

- [x] I have updated the documentation

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/helm-values/types.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
-export type HelmDockerImageDependency = {
+type HDIDBasic = {
```
Comment (viceice) [inline lib/modules/manager/helm-values/types.ts:1]: 👎 Use long names, prefer interfaces and export public types

**Review by viceice — CHANGES_REQUESTED** [review]
please check git author / commiter (use force push to update, no new pr)

File: lib/modules/manager/helm-values/types.ts (hunk 1 above)
Comment (karl-johan-grahn) [reply lib/modules/manager/helm-values/types.ts]: Thanks, should be fixed now

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs lint fix

**Review by viceice — COMMENTED** [review]
needs fixes

**Review by viceice — COMMENTED** [review]
last commit by wrong git author, needs fix and force-push

**Comment by karl-johan-grahn** [comment]
Fixed, please rerun lint and test

File: lib/modules/manager/helm-values/extract.ts [hunk 2]
```diff
@@ -46,7 +46,8 @@ function findDependencies(
       let registry = currentItem.registry;
       registry = registry ? `${registry}/` : '';
       const repository = String(currentItem.repository);
-      const tag = String(currentItem.tag);
+      let tag = currentItem.tag;
+      tag = tag ? String(tag) : String(currentItem.version);
```
Comment (viceice) [inline lib/modules/manager/helm-values/extract.ts:50]:
```suggestion
      const tag = `${currentItem.tag ?? currentItem.version}`;
```
simplify

File: lib/modules/manager/helm-values/types.ts [hunk 3]
```diff
@@ -1,5 +1,14 @@
-export type HelmDockerImageDependency = {
+interface HelmDockerImageDependencyBasic {
```
Comment (viceice) [inline lib/modules/manager/helm-values/types.ts:1]:
```suggestion
export interface HelmDockerImageDependencyBasic {
```

always export public types, even it it's only referenced indirectly

File: lib/modules/manager/helm-values/types.ts [hunk 4]
```diff
@@ -1,5 +1,14 @@
...
   registry?: string;
   repository: string;
+}
+interface HelmDockerImageDependencyTag {
```
Comment (viceice) [inline lib/modules/manager/helm-values/types.ts:5]:
```suggestion
export interface HelmDockerImageDependencyTag extends HelmDockerImageDependencyBasic {
```

File: lib/modules/manager/helm-values/types.ts [hunk 5]
```diff
@@ -1,5 +1,14 @@
...
   tag: string;
-};
+  version?: never;
+}
+interface HelmDockerImageDependencyVersion {
+  version: string;
+  tag?: never;
+}
+export type HelmDockerImageDependency = HelmDockerImageDependencyBasic &
+  (HelmDockerImageDependencyTag | HelmDockerImageDependencyVersion);
```
Comment (viceice) [inline lib/modules/manager/helm-values/types.ts:14]:
```suggestion
export type HelmDockerImageDependency = HelmDockerImageDependencyTag | HelmDockerImageDependencyVersion;
```

File: lib/modules/manager/helm-values/extract.ts (hunk 2 above)
Comment (viceice) [reply lib/modules/manager/helm-values/extract.ts:50]: We should validate, that the yaml parser extracted that as strings and log a warning if it's a number.

If  `tag: 1.2.10` then the yaml parser will read `1.2.1` as number (yaml standard).

Can be done in seperate PR

Comment (karl-johan-grahn) [reply lib/modules/manager/helm-values/extract.ts:50]: `tag` is the same as before, what is added here is the ability to parse the version from a `version` field

Comment (viceice) [reply lib/modules/manager/helm-values/extract.ts:50]: please apply my suggestion

Comment (karl-johan-grahn) [reply lib/modules/manager/helm-values/extract.ts]: done

**Review by viceice — DISMISSED** [review]

**Comment by karl-johan-grahn** [comment]
I don't know what the lint error is about, when I run `yarn lint-fix` locally it doesn't complain about `lib/modules/manager/helm-values/types.ts`

File: lib/modules/manager/helm-values/types.ts [hunk 6]
```diff
@@ -1,5 +1,14 @@
-export type HelmDockerImageDependency = {
+export interface HelmDockerImageDependencyBasic {
   registry?: string;
   repository: string;
+}
+export interface HelmDockerImageDependencyTag extends HelmDockerImageDependencyBasic {
```
Comment (viceice) [inline lib/modules/manager/helm-values/types.ts:5]:
```suggestion

export interface HelmDockerImageDependencyTag extends HelmDockerImageDependencyBasic {
```

File: lib/modules/manager/helm-values/types.ts [hunk 7]
```diff
@@ -1,5 +1,14 @@
...
   tag: string;
-};
+  version?: never;
+}
+export interface HelmDockerImageDependencyVersion extends HelmDockerImageDependencyBasic {
```
Comment (viceice) [inline lib/modules/manager/helm-values/types.ts:9]:
```suggestion

export interface HelmDockerImageDependencyVersion extends HelmDockerImageDependencyBasic {
```

File: lib/modules/manager/helm-values/types.ts [hunk 8]
```diff
@@ -1,5 +1,14 @@
...
+  version: string;
+  tag?: never;
+}
+export type HelmDockerImageDependency = HelmDockerImageDependencyTag |
```
Comment (viceice) [inline lib/modules/manager/helm-values/types.ts:13]:
```suggestion

export type HelmDockerImageDependency = HelmDockerImageDependencyTag |
```

**Comment by karl-johan-grahn** [comment]
Alright here we go again, ready for lint and test again

**Comment by viceice** [comment]
> Alright here we go again, ready for lint and test again

👉 https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#re-requesting-a-review

**Review by viceice — DISMISSED** [review]

**Comment by karl-johan-grahn** [comment]
Once again lint fails for `types.ts` but the build log doesn't help anything and running `yarn lint-fix` locally doesn't report anything wrong either

**Review by viceice — APPROVED** [review]
you probably didn't updated your local node deps, run `yarn install` then you see changes like i pushed

**Review by rarkins — APPROVED** [review]

**Pull request merged**